packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,398 @@
+                <>
+                  <SearchBoxInput {...inputProps} />
+                  <SearchBoxCloseButton {...closeButtonProps} />
+                </>
+              )}
+            </div>
+          </>
+        )}
+      </InputFlexContainer>
+    </SearchBoxContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+export type SearchBoxProps = PolymorphicComponentProps<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+// ----------------------------------------------------------------------------
+export type SearchBoxIconProps = PolymorphicComponentProps<'span', IconProps>;
+
+const SearchBoxIcon = React.forwardRef((props, ref) => {
+  const { size, className, children, ...rest } = props;
+  const { size: sizeContext } = useSafeContext(SearchBoxContext);
+
+  return (
+    <Icon
+      aria-hidden
+      className={cx('iui-svg-icon', 'iui-search-icon', className)}
+      size={size ?? sizeContext}
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <SvgSearch />}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', IconProps>;
+
+// ----------------------------------------------------------------------------
+
+export type SearchBoxInputProps = {
+  label?: string;
+} & React.ComponentProps<'input'>;
+
+const SearchBoxInput = React.forwardRef(
+  (props: SearchBoxInputProps, ref: React.Ref<HTMLInputElement>) => {
+    const {
+      className,
+      label = 'Search input',
+      id: idProp,
+      disabled: disabledProp,
+      ...rest
+    } = props;
+
+    const { inputId, setInputId, disabled, inputRef } =
+      useSafeContext(SearchBoxContext);
+
+    if (idProp && idProp !== inputId) {
+      setInputId(idProp);
+    }
+
+    return (
+      <input
+        id={idProp ?? inputId}
+        aria-label={label}
+        ref={useMergedRefs(ref, inputRef)}
+        type='text'
+        className={cx('iui-search-input', className)}
+        disabled={disabledProp ?? disabled}
+        {...rest}
+      />
+    );
+  },
+);
+
+// ----------------------------------------------------------------------------
+
+/**
+ * SearchBox.Button component to add to your SearchBox.
+ */
+export type SearchBoxButtonProps = PolymorphicComponentProps<
+  'button',
+  IconButtonProps
+>;
+
+const SearchBoxButton = React.forwardRef((props, ref) => {
+  const { children, size, disabled: disabledProp, ...rest } = props;
+  const { size: sizeContext, disabled } = useSafeContext(SearchBoxContext);
+
+  return (
+    <IconButton
+      styleType='borderless'
+      size={size ?? sizeContext}
+      ref={ref}
+      disabled={disabledProp ?? disabled}
+      {...rest}
+    >
+      {children ?? <SvgSearch />}
+    </IconButton>
+  );
+}) as PolymorphicForwardRefComponent<'button', IconButtonProps>;
+
+// ----------------------------------------------------------------------------
+
+/**
+ * SearchBox.Button component to add to your SearchBox.
+ */
```

Comment:
these jsdoc comments above types are unnecessary

Reply:
Moved jsdoc comments a bit.

Reply:
it's not needed in the components either, because it's already present inside Object.assign at the end

Reply:
Removed.